network/builder/src/dummy.rs

```diff
@@ -130,13 +130,13 @@ pub fn setup_network() -> DummyNetwork {
     let dialer_addr: NetworkAddress = "/ip4/127.0.0.1/tcp/0".parse().unwrap();
     let listener_addr: NetworkAddress = "/ip4/127.0.0.1/tcp/0".parse().unwrap();
 
-    // Setup trusted peers.
-    let seed_pubkeys: HashMap<_, _> = vec![
-        (dialer_peer_id, dialer_pubkeys),
-        (listener_peer_id, listener_pubkeys),
-    ]
-    .into_iter()
-    .collect();
+    // Setup seed peers
+    let mut seeds = TrustedPeerSet::new();
+    seeds.insert(
+        dialer_peer_id,
+        TrustedPeer::new(vec![dialer_addr.clone()], dialer_pubkeys),
```

Comment:
I don't think you want `dialer_addr` here, since it doesn't include the pubkey or handshake protocols.